### PR TITLE
Support subcolumns

### PR DESCRIPTION
**Pull request by billy-addepar**
- Support subcolumns in the header. As a start, we only support 1 layer of subcolumn.
- The `bodyColumns` refers to the `columns` used in the body for rows. In normal case, it's the same as `columns`. If header has subcolumns, it's the concatenation of all `subcolumns.`

screen shot 2017-11-15 at 4 17 47 pm

### CODE REVIEW

**Comment by billy-addepar** [comment]
@Addepar/ice

File: addon/components/ember-table-base-cell.js [hunk 1]
```diff
@@ -39,7 +40,14 @@ export default class EmberTableCell extends Component {
...
   scheduleSync() {
     scheduler.schedule('sync', () => {
-      const { height } = this.element.getBoundingClientRect();
+      let { height } = this.element.getBoundingClientRect();
+
+      if (this.get('isFixed')) {
+        // Get the max height of the div element height and other cells height in case other cells
+        // are empty.
+        const fixedDivElement = this.element.getElementsByClassName('is-fixed')[0];
+        height = Math.max(height, fixedDivElement.getBoundingClientRect().height);
```
Comment (pzuraq) [inline addon/components/ember-table-base-cell.js]: @billy-addepar what happens if the user changes the row height or contents, and the row should shrink in height?

File: addon/components/ember-table-base-cell.js [hunk 2]
```diff
@@ -39,7 +40,14 @@ export default class EmberTableCell extends Component {
 
   scheduleSync() {
     scheduler.schedule('sync', () => {
-      const { height } = this.element.getBoundingClientRect();
+      let { height } = this.element.getBoundingClientRect();
+
+      if (this.get('isFixed')) {
```
Comment (pzuraq) [inline addon/components/ember-table-base-cell.js]: We should revert this commit until we're ready to fully address the issues of keeping the row height and cell height in sync. In the meantime, we can opt into the performance improvement of using fixed heights.

**Comment by billy-addepar** [comment]
@Addepar/ice This is ready for review again.

**Pull request merged**